homeassistant/helpers/config_validation.py

```diff
@@ -147,6 +147,18 @@ def entity_ids(value: Union[str, Sequence]) -> Sequence[str]:
     return [entity_id(ent_id) for ent_id in value]
 
 
+def entities_domain(domain: str):
+    """Validate that entities belong to domain."""
+    def validate(values: Union[str, Sequence]) -> Sequence[str]:
+        """Test if entitiy domain is domain."""
+        values = entity_ids(values)
+        for ent_id in values:
+            if split_entity_id(ent_id)[0] != domain:
+                raise vol.Invalid("Entity ID does not belong to domain")
```

Comment:
From my past experience voluptuous is not that great at reporting where exactly a validation error occured, and having a message like this can be quite cryptic without context. This is only a suggestion: could you maybe also print out the entity_id and the domain it's supposed to be in?